processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java

```diff
@@ -133,9 +138,16 @@ public int getPriority() {
         List<String> variableNames = new LinkedList<>();
 
         for ( TypeMirror usedMapper : mapperAnnotation.uses() ) {
+            boolean isSingleton = typeUtils.asElement( usedMapper ).getEnclosedElements().stream()
+            .filter( a -> a.getKind().isField() )
+            .filter( a -> a.getModifiers().containsAll( Arrays.asList( PUBLIC, STATIC, FINAL ) ) )
+            .filter( a -> a.getSimpleName().contentEquals( "INSTANCE" ) )
+            .filter( a -> typeUtils.isSameType( a.asType(), usedMapper ) )
+            .anyMatch( a -> true );
```

Comment:
I don't really like that we have so many different method calls here.

I would suggest the following. Extract a method named `hasInstanceMapper` that takes `Element` as a parameter and returns `true` if  these 4 conditions here are satisfied. Additionally instead of using filter you can directly use `anyMatch(this::hasInstanceMapper)`. 

I would also suggest extracting the `Arrays.asList` into a static field in the `MapperCreationProcessor` in order to avoid creating that list all the time.

Reply:
I just pushed changes to extract  methods/constants (more or less) as suggested.

Or do you prefer a rebase/squash ?

Reply:
There is some conflict now, if you can rebase your changes on top of master it would branch, it would be great. If you can't we can do it when merging

Reply:
I have rebased th MR on top os master.